chat-server/src/routes/conversations/conversationsRouter.ts

```diff
@@ -77,6 +114,43 @@ export interface ConversationsRouterParams {
   rateLimitConfig?: ConversationsRateLimitConfig;
   findContent: FindContentFunc;
   makeReferenceLinks?: MakeReferenceLinksFunc;
+
+  /**
+    Middleware to put in front of all the routes in the conversationsRouter.
+    You can use this to do things like authentication, data validation, etc.
+
+    If you want the middleware to run only on certain routes,
+    you can add conditional logic inside the middleware. For example:
+
+    ```ts
+    const someMiddleware: ConversationsMiddleware = (req, res, next) => {
+      if (req.path === "/conversations") {
+        // Do something
+      }
+      next();
+    }
+    ```
+
+   */
+  middleware?: ConversationsMiddleware[];
+
+  /**
+    Function that takes the request + response and returns any custom data you want to include
+    in the {@link Conversation} persisted to the database.
+    For example, you might want to store the user's email address with the conversation.
+    The custom data is persisted to the database with the Conversation in the
+    {@link Conversation.customData} field.
+   */
+  createConversationCustomData?: AddCustomDataFunc;
```

Comment:
I don't think it's a blocker, but it feels a bit weird to me to use these route-specific custom data middleware slots. Maybe in the future we can have a more generic "post route" middleware parameter that filters on `req` like your example code for `middleware`?